src/test/app/Batch_test.cpp

```diff
@@ -0,0 +1,2181 @@
+                ter(tesSUCCESS));
+            auto const txIDs = env.tx()->getFieldV256(sfTransactionIDs);
+            TxID const batchId = env.tx()->getTransactionID();
+            std::vector<TestBatchData> testCases = {
+                {"tesSUCCESS", to_string(txIDs[0])},
+                {"tesSUCCESS", to_string(txIDs[1])},
+            };
+            env.close();
+            validateBatch(env, batchId, testCases);
+
+            // Alice consumes sequences (# of txns)
+            BEAST_EXPECT(env.seq(alice) == seq + 3);
+
+            // Alice pays XRP & Fee
+            BEAST_EXPECT(env.balance(alice) == preAlice - XRP(2) - batchFee);
+
+            // Bob receives XRP
+            BEAST_EXPECT(env.balance(bob) == preBob + XRP(2));
+        }
+
+        // nothing
+        {
+            test::jtx::Env env{*this, envconfig()};
+
+            auto const alice = Account("alice");
+            auto const bob = Account("bob");
+            auto const carol = Account("carol");
+            env.fund(XRP(1000), alice, bob, carol);
+            env.close();
+
+            auto const preAlice = env.balance(alice);
+            auto const preBob = env.balance(bob);
+
+            auto const batchFee = calcBatchFee(env, 0, 2);
+            auto const seq = env.seq(alice);
+            env(batch::batch(alice, seq, batchFee, tfAllOrNothing),
+                batch::add(pay(alice, bob, XRP(1)), seq + 1),
+                batch::add(pay(alice, bob, XRP(999)), seq + 2),
+                ter(tesSUCCESS));
+            auto const txIDs = env.tx()->getFieldV256(sfTransactionIDs);
+            TxID const batchId = env.tx()->getTransactionID();
+            std::vector<TestBatchData> testCases = {};
+            env.close();
+            validateBatch(env, batchId, testCases);
+
+            // Alice consumes sequence
+            BEAST_EXPECT(env.seq(alice) == seq + 1);
+
+            // Alice pays Fee
+            BEAST_EXPECT(env.balance(alice) == preAlice - batchFee);
+
+            // Bob should not be affected
+            BEAST_EXPECT(env.balance(bob) == preBob);
+        }
+    }
+
+    void
+    testOnlyOne(FeatureBitset features)
+    {
+        testcase("only one");
+
+        using namespace test::jtx;
+        using namespace std::literals;
+
+        test::jtx::Env env{*this, envconfig()};
+
+        auto const alice = Account("alice");
+        auto const bob = Account("bob");
+        auto const carol = Account("carol");
+        env.fund(XRP(1000), alice, bob, carol);
+        env.close();
+
+        auto const preAlice = env.balance(alice);
+        auto const preBob = env.balance(bob);
+
+        auto const batchFee = calcBatchFee(env, 0, 3);
+        auto const seq = env.seq(alice);
+        env(batch::batch(alice, seq, batchFee, tfOnlyOne),
+            batch::add(pay(alice, bob, XRP(999)), seq + 1),
+            batch::add(pay(alice, bob, XRP(1)), seq + 2),
+            batch::add(pay(alice, bob, XRP(1)), seq + 3),
```

Comment:
nit: doesn't change the test behavior, prefer to change the amount of the unprocessed txn to be different from the previous txn 

```suggestion
            batch::add(pay(alice, bob, XRP(2)), seq + 3),
```

Reply:
fixed